pkg/catalog/catalog.go

```diff
@@ -45,18 +53,42 @@ func GetDevfileRegistries(registryName string) (map[string]string, error) {
 	return devfileRegistries, nil
 }
 
+const indexPath = "/devfiles/index.json"
+
 // GetDevfileIndex loads the devfile registry index.json
-func GetDevfileIndex(devfileIndexLink string) ([]DevfileIndexEntry, error) {
-	var devfileIndex []DevfileIndexEntry
+func GetDevfileIndex(registryName string) ([]DevfileIndexEntry, error) {
+	registries, err2 := GetDevfileRegistries(registryName)
+	if err2 != nil {
+		return nil, err2
+	}
+
+	registry := registries[registryName]
+	return getDevfileIndexEntries(registry)
+}
+
+func getDevfileIndexEntriesFrom(registryName, registryURL string) ([]DevfileIndexEntry, error) {
```

Comment:
some comments

Reply:
> some comments. Also this function seems to be only used in tests

So? What would you prefer? Have a function only used for tests or no tests?

Reply:
> So? What would you prefer? Have a function only used for tests or no tests?

I mean do we even need this function? There is no usage of this function in the code except the test written for it. It also calls `getDevfileIndexEntries` internally. Why not test `getDevfileIndexEntries` instead and remove this unused function?